packages/tailwindcss/src/ast.ts

```diff
@@ -48,17 +48,17 @@ export function objectToAst(obj: CssInJs): AstNode[] {
   let ast: AstNode[] = []
 
   for (let [name, value] of Object.entries(obj)) {
-    if (typeof value === 'string') {
+    if (typeof value !== 'object') {
       if (!name.startsWith('--') && value === '@slot') {
         ast.push(rule(name, [rule('@slot', [])]))
       } else {
         // Convert camelCase to kebab-case:
         // https://github.com/postcss/postcss-js/blob/b3db658b932b42f6ac14ca0b1d50f50c4569805b/parser.js#L30-L35
         name = name.replace(/([A-Z])/g, '-$1').toLowerCase()
 
-        ast.push(decl(name, value))
+        ast.push(decl(name, String(value)))
```

Comment:
I wonder if we could do this casting in the plugin API somewhere instead of here, so that core can always assume that values are strings?

Reply:
`objectToAst` is only used by the plugin API so it still seems reasonable for it to be in that function. Perhaps we can move the fn though so it still feels self contained.

Reply:
Yep okay that's cool!

Reply:
Done.